cli/src/commands/operation.rs

```diff
@@ -395,14 +460,541 @@ fn cmd_op_abandon(
+            )?
+        }
+    } else {
+        vec![op_walk::resolve_op_for_load(repo_loader, head_op_str)?]
+    };
+    let current_op_id = match &*head_ops {
+        [op] => Some(op.id()),
+        _ => None,
+    };
+    let op = op_walk::resolve_op_for_load(repo_loader, &args.operation)?;
+    let parent_op = merge_operations(repo_loader, command.settings(), op.parents())?;
+    if parent_op.is_none() {
+        return Err(user_error("Cannot show the root operation"));
+    }
+    let parent_op = parent_op.unwrap();
+    let with_content_format = LogContentFormat::new(ui, command.settings())?;
+
+    // TODO: Should we make this customizable via clap arg?
+    let template;
+    {
+        let language = OperationTemplateLanguage::new(
+            repo_loader.op_store().root_operation_id(),
+            current_op_id,
+            command.operation_template_extensions(),
+        );
+        let text = command.settings().config().get_string("templates.op_log")?;
+        template = command
+            .parse_template(
+                ui,
+                &language,
+                &text,
+                OperationTemplateLanguage::wrap_operation,
+            )?
+            .labeled("op_log");
+    }
+
+    let parent_repo = repo_loader.load_at(&parent_op)?;
+    let repo = repo_loader.load_at(&op)?;
+
+    ui.request_pager();
+    template.format(&op, ui.stdout_formatter().as_mut())?;
+    writeln!(ui.stdout())?;
+
+    show_op_diff(
+        ui,
+        command,
+        &parent_repo,
+        &repo,
+        !args.no_graph,
+        &with_content_format,
+        &args.diff_format,
+        args.patch,
+    )
+}
+
+fn cmd_op_diff(
+    ui: &mut Ui,
+    command: &CommandHelper,
+    args: &OperationDiffArgs,
+) -> Result<(), CommandError> {
+    // TODO: Should we load the repo here?
+    let workspace = command.load_workspace()?;
+    let repo_loader = workspace.repo_loader();
+    let head_op_str = &command.global_args().at_operation;
+    let from_op;
+    let to_op;
+    if args.from.is_some() || args.to.is_some() {
+        from_op =
+            op_walk::resolve_op_for_load(repo_loader, args.from.as_ref().unwrap_or(head_op_str))?;
+        to_op = op_walk::resolve_op_for_load(repo_loader, args.to.as_ref().unwrap_or(head_op_str))?;
+    } else {
+        to_op = op_walk::resolve_op_for_load(
+            repo_loader,
+            args.operation.as_ref().unwrap_or(head_op_str),
+        )?;
+        let merged_parents_op = merge_operations(repo_loader, command.settings(), to_op.parents())?;
+        if merged_parents_op.is_none() {
+            return Err(user_error("Cannot diff operation with no parents"));
+        }
+        from_op = merged_parents_op.unwrap();
+    }
+    let with_content_format = LogContentFormat::new(ui, command.settings())?;
+
+    let from_repo = repo_loader.load_at(&from_op)?;
+    let to_repo = repo_loader.load_at(&to_op)?;
+
+    ui.request_pager();
```

Comment:
This looked slightly confusing when I used it. I would format it a little differently:

"Changes in operation abcd: squash commits
Parent operation xyzw: ..."

Also, colors are not necessary, but would be nice. I guess the description should just get `op_log description` label, and the id should bet `op_log id` according to https://github.com/martinvonz/jj/blob/4cca84515a1c18a221cb2ee31a036bde60a3db2f/cli/src/config/colors.toml#L86.